### PR TITLE
CR Oracle + Timelock Ops

### PR DESCRIPTION
**Pull request by Joeysantoro**
# PR Type: Feature

### PR Description:
Executes a few cleanup actions related to collateralization and operations:
  1. Add a collateralization FEI lens to the FEI in the tribal council timelock and old rari timelocks
  2. Accept beneficiary of the old rari timelocks to tribal council timelock.  
  3. Remove deprecated namedStaticPCVDeposit from collateralization oracle. 
  4. Trigger new ETH LBP to convert 10k ETH to DAI.
  5. Grant tribal council safe the SWAP_ADMIN_ROLE to initiate LBPs after governance funds them.

### PR Checklist - Feature (Proposal)

- [x] All Tests Passing
- [x] Proposal Added to ProposalsConfig
- [x] Fork Block Correct
- [x] Remove Any .only's on Tests
- [x] Update Documentation If Needed
- [x] Update Roles Config
- [x] Proposal Submitted

### CODE REVIEW

**Comment by Joeysantoro** [comment]
EDIT: deleted old calldata

**Review by xklob — DISMISSED** [review]

File: proposals/dao/timelock_admin.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// Tears down any changes made in setup() that need to be
+// cleaned up before doing any validation checks.
+const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in teardown for fip${fipNumber}`);
+};
+
+// Run any validations required on the fip using mocha or console logging
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log('here');
```
Comment (thomas-waite) [inline proposals/dao/timelock_admin.ts]: Nit: Remove this `console.log`

File: proposals/dao/timelock_admin.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { expect } from 'chai';
+import hre from 'hardhat';
+
+const fipNumber = 'timelock_admin';
+
+// Do any deployments
+// This should exclusively include new contract deployments
+const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  const factory = (await hre.ethers.getContractFactory('ERC20PCVDepositWrapper')) as ERC20PCVDepositWrapper__factory;
+  const rariTimelockFeiOldLens = await factory.deploy(addresses.rariInfraFeiTimelock, addresses.fei, true);
```
Comment (thomas-waite) [inline proposals/dao/timelock_admin.ts]: Will make sure to add lenses for contracts with significant Fei that are becoming part of the Fei ecosystem in the future!

File: proposals/description/timelock_admin.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { TemplatedProposalDescription } from '@custom-types/types';
+
+const timelock_admin: TemplatedProposalDescription = {
+  title: 'Timelock Updates',
```
Comment (thomas-waite) [inline proposals/description/timelock_admin.ts]: Would probably add a TIP number?

File: protocol-configuration/proposalsConfig.ts [hunk 4]
```diff
@@ -3,49 +3,59 @@ import tip_111 from '@proposals/description/tip_111';
 import pod_exec_v2 from '@proposals/description/pod_exec_v2';
 import tokemak_withdraw from '@proposals/description/tokemak_withdraw';
 import tip_112 from '@proposals/description/tip_112';
+import timelock_admin from '@proposals/description/timelock_admin';
 
 const proposals: TemplatedProposalsConfigMap = {
-  tip_111: {
+  // tip_111: {
```
Comment (thomas-waite) [inline protocol-configuration/proposalsConfig.ts]: Think this is leftover commenting and needs deleting? `tokemak_withdraw` and `tip_112` can both be removed.

`pod_exec_v2` needs to stay here, it hasn't been proposed/executed by the TC yet

**Review by thomas-waite — COMMENTED** [review]
Looks good overall, few tidy up comments

File: proposals/description/timelock_admin.ts (hunk 3 above)
Comment (Joeysantoro) [reply proposals/description/timelock_admin.ts]: Yup!

File: test/integration/tests/buybacks.ts [hunk 5]
```diff
@@ -130,57 +130,4 @@ describe('e2e-buybacks', function () {
...
-      expect(await collateralizationOracleWrapper.isExceededDeviationThreshold()).to.be.false;
-
-      // UpdateIfOutdated succeeds
-      await collateralizationOracleWrapper.updateIfOutdated();
-
-      expect(await collateralizationOracleWrapper.isOutdatedOrExceededDeviationThreshold()).to.be.false;
-
-      // Increase PCV balance to exceed deviation threshold
-      const pcvStats = await collateralizationOracleWrapper.pcvStats();
-      await namedStaticPCVDepositWrapper.addDeposit({
```
Comment (Joeysantoro) [inline test/integration/tests/buybacks.ts:163]: this test relies on named static PCV deposit which will be deprecated in this proposal. Erwan's #923 also deprecates the keeper

File: protocol-configuration/proposalsConfig.ts [hunk 6]
```diff
@@ -3,59 +3,59 @@ import tip_111 from '@proposals/description/tip_111';
...
+    affectedContractSignoff: [
+      /*'dpiToDaiLBPSwapper',
+      'compoundDaiPCVDeposit',
+      'tribalCouncilSafe',
+      'ratioPCVControllerV2'*/
+    ],
+    deprecatedContractSignoff: [],
+    category: ProposalCategory.DAO
+  },
+  tip_112: {
```
Comment (thomas-waite) [inline protocol-configuration/proposalsConfig.ts]: You can remove tip_112 here if you want - it's the hack repayment DAO vote, wasn't executed and no longer needed. Removing will speed up your e2es

File: protocol-configuration/proposalsConfig.ts [hunk 7]
```diff
@@ -3,59 +3,59 @@ import tip_111 from '@proposals/description/tip_111';
...
+    deploy: false, // deploy flag for whether to run deploy action during e2e tests or use mainnet state
+    totalValue: 0, // amount of ETH to send to DAO execution
+    proposal: tip_112, // full proposal file, imported from '@proposals/description/fip_xx.ts'
+    proposalId: '',
+    affectedContractSignoff: ['core', 'fuseFixer', 'pcvGuardianNew'],
+    deprecatedContractSignoff: [],
+    category: ProposalCategory.DAO
+  },
+  tokemak_withdraw: {
```
Comment (thomas-waite) [inline protocol-configuration/proposalsConfig.ts]: `tokemak_withdraw` can be removed as already executed on-chain - will speed up your e2es. You might need to update the block number to reflect that

File: proposals/description/tip_115.ts [hunk 8]
```diff
@@ -51,9 +51,60 @@ const tip_115: TemplatedProposalDescription = {
...
+      method: 'swap()',
+      arguments: (addresses) => [],
+      description: 'Trigger new ETH LBP swap'
+    },
+    {
+      target: 'core',
+      values: '0',
+      method: 'grantRole(bytes32,address)',
+      arguments: (addresses) => [
+        '0x471cfe1a44bf1b786db7d7104d51e6728ed7b90a35394ad7cc424adf8ed16816', // SWAP_ADMIN_ROLE
```
Comment (thomas-waite) [inline proposals/description/tip_115.ts:95]: Nit, you can import `ethers` now and replace this with `ethers.utils.id('SWAP_ADMIN_ROLE)`

File: proposals/dao/tip_115.ts [hunk 9]
```diff
@@ -47,6 +52,9 @@ const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts,
   expect(await contracts.rariTimelockFeiOldLens.balanceReportedIn()).to.be.equal(addresses.fei);
   expect(await contracts.tribalCouncilTimelockFeiLens.balanceReportedIn()).to.be.equal(addresses.fei);
   expect(await contracts.namedStaticPCVDepositWrapper.numDeposits()).to.be.equal('0');
+  expect(await contracts.ethToDaiLBPSwapper.swapEndTime()).to.be.gt(
+    ethers.BigNumber.from((await time.latest()).toString())
+  );
```
Comment (thomas-waite) [inline proposals/dao/tip_115.ts:57]: Depends how confident are, but I would probably make sure you can do a swap via the LBP pool here

File: proposals/description/tip_115.ts [hunk 10]
```diff
@@ -51,9 +51,60 @@ const tip_115: TemplatedProposalDescription = {
...
+      values: '0',
+      method: 'withdrawToSafeAddress(address,address,uint256,bool,bool)',
+      arguments: (addresses) => [
+        addresses.ethPSM,
+        addresses.ethToDaiLBPSwapper,
+        '10000000000000000000000',
+        false,
+        false
+      ],
+      description: 'Withdraw 10k ETH to LBP Swapper'
```
Comment (thomas-waite) [inline proposals/description/tip_115.ts:81]: Weights are 95 / 5 . You're sending $1m DAI and ~$10.5m ETH. So overfunding by $500k, should be fine

**Review by thomas-waite — COMMENTED** [review]
Looks fine to me, will give a final once over tomorrow then approve

**Comment by Joeysantoro** [comment]
Calldata: 
```
0x8f2a0bb000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000000000024000000000000000000000000000000000000000000000000000000000000003c0000000000000000000000000000000000000000000000000000000000000000081216f2d37b0a60a2451f16caffab42bb8fa9a8e67d1daa520e241e3f85a61180000000000000000000000000000000000000000000000000000000000054600000000000000000000000000000000000000000000000000000000000000000b000000000000000000000000ff6f59333cfd8f4ebc14ad0a0e181a83e655d257000000000000000000000000ff6f59333cfd8f4ebc14ad0a0e181a83e655d257000000000000000000000000ff6f59333cfd8f4ebc14ad0a0e181a83e655d25700000000000000000000000006dacca04e201ad31393754e68da04dc14778fa600000000000000000000000006dacca04e201ad31393754e68da04dc14778fa6000000000000000000000000fafc562265a49975e8b20707eac966473795cf90000000000000000000000000625cf6aa7dafb154f3eb6be87592110e30290dee00000000000000000000000002435948f84d7465fb71de45aba6098fc6ec299300000000000000000000000002435948f84d7465fb71de45aba6098fc6ec2993000000000000000000000000f7991f4698ffb6716982aec7f78964dd731c4a540000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b9000000000000000000000000000000000000000000000000000000000000000b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000b000000000000000000000000000000000000000000000000000000000000016000000000000000000000000000000000000000000000000000000000000001c00000000000000000000000000000000000000000000000000000000000000220000000000000000000000000000000000000000000000000000000000000028000000000000000000000000000000000000000000000000000000000000002e00000000000000000000000000000000000000000000000000000000000000340000000000000000000000000000000000000000000000000000000000000038000000000000000000000000000000000000000000000000000000000000003c000000000000000000000000000000000000000000000000000000000000004a0000000000000000000000000000000000000000000000000000000000000058000000000000000000000000000000000000000000000000000000000000005c00000000000000000000000000000000000000000000000000000000000000024dea02892000000000000000000000000614d46b7eb2ac1a359b8835d64954f3ee4e6f676000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000024dea028920000000000000000000000004037a70152f4c88ad40522f35bd4ddd17e6b20520000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000240a633e0900000000000000000000000006dacca04e201ad31393754e68da04dc14778fa60000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000244d6d199100000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000244d6d199100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044929e1620000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044929e1620000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a4fe68d76e0000000000000000000000002a188f9eb761f70ecea083ba6c2a40145078dfc2000000000000000000000000f7991f4698ffb6716982aec7f78964dd731c4a5400000000000000000000000000000000000000000000d3c21bcecceda1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a4fe68d76e00000000000000000000000098e5f5706897074a4664dd3a32eb80242d6e694b000000000000000000000000f7991f4698ffb6716982aec7f78964dd731c4a5400000000000000000000000000000000000000000000021e19e0c9bab2400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000048119c0650000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000442f2ff15d471cfe1a44bf1b786db7d7104d51e6728ed7b90a35394ad7cc424adf8ed168160000000000000000000000002ec598d8e3df35e5d6f13ae2f05a7bb2704e92ea00000000000000000000000000000000000000000000000000000000
```
Execute Calldata: 
```
0xe38335e500000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000022000000000000000000000000000000000000000000000000000000000000003a0000000000000000000000000000000000000000000000000000000000000000081216f2d37b0a60a2451f16caffab42bb8fa9a8e67d1daa520e241e3f85a6118000000000000000000000000000000000000000000000000000000000000000b000000000000000000000000ff6f59333cfd8f4ebc14ad0a0e181a83e655d257000000000000000000000000ff6f59333cfd8f4ebc14ad0a0e181a83e655d257000000000000000000000000ff6f59333cfd8f4ebc14ad0a0e181a83e655d25700000000000000000000000006dacca04e201ad31393754e68da04dc14778fa600000000000000000000000006dacca04e201ad31393754e68da04dc14778fa6000000000000000000000000fafc562265a49975e8b20707eac966473795cf90000000000000000000000000625cf6aa7dafb154f3eb6be87592110e30290dee00000000000000000000000002435948f84d7465fb71de45aba6098fc6ec299300000000000000000000000002435948f84d7465fb71de45aba6098fc6ec2993000000000000000000000000f7991f4698ffb6716982aec7f78964dd731c4a540000000000000000000000008d5ed43dca8c2f7dfb20cf7b53cc7e593635d7b9000000000000000000000000000000000000000000000000000000000000000b00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000b000000000000000000000000000000000000000000000000000000000000016000000000000000000000000000000000000000000000000000000000000001c00000000000000000000000000000000000000000000000000000000000000220000000000000000000000000000000000000000000000000000000000000028000000000000000000000000000000000000000000000000000000000000002e00000000000000000000000000000000000000000000000000000000000000340000000000000000000000000000000000000000000000000000000000000038000000000000000000000000000000000000000000000000000000000000003c000000000000000000000000000000000000000000000000000000000000004a0000000000000000000000000000000000000000000000000000000000000058000000000000000000000000000000000000000000000000000000000000005c00000000000000000000000000000000000000000000000000000000000000024dea02892000000000000000000000000614d46b7eb2ac1a359b8835d64954f3ee4e6f676000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000024dea028920000000000000000000000004037a70152f4c88ad40522f35bd4ddd17e6b20520000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000240a633e0900000000000000000000000006dacca04e201ad31393754e68da04dc14778fa60000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000244d6d199100000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000244d6d199100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044929e1620000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000044929e1620000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a4fe68d76e0000000000000000000000002a188f9eb761f70ecea083ba6c2a40145078dfc2000000000000000000000000f7991f4698ffb6716982aec7f78964dd731c4a5400000000000000000000000000000000000000000000d3c21bcecceda1000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a4fe68d76e00000000000000000000000098e5f5706897074a4664dd3a32eb80242d6e694b000000000000000000000000f7991f4698ffb6716982aec7f78964dd731c4a5400000000000000000000000000000000000000000000021e19e0c9bab2400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000048119c0650000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000442f2ff15d471cfe1a44bf1b786db7d7104d51e6728ed7b90a35394ad7cc424adf8ed168160000000000000000000000002ec598d8e3df35e5d6f13ae2f05a7bb2704e92ea00000000000000000000000000000000000000000000000000000000
```

**Review by thomas-waite — APPROVED** [review]
Looks okay to me. I would add a validation that you can do a swap and that the implied price is reasonable first if there's time. 

Have verified the propose and execute calldata, looks good

**Pull request merged**